ddp-workspace/projects/ddp-sdk/src/lib/services/activity/validators/activityFileValidationRule.ts

```diff
@@ -0,0 +1,36 @@
+import { ActivityAbstractValidationRule } from './activityAbstractValidationRule';
+import { ActivityFileQuestionBlock } from '../../../models/activity/activityFileQuestionBlock';
+import { ActivityFileAnswerDto } from '../../../models/activity/activityFileAnswerDto';
+
+export class ActivityFileValidationRule extends ActivityAbstractValidationRule {
+    constructor(private block: ActivityFileQuestionBlock) {
+        super(block);
+    }
+
+    public recalculate(answerBeforeUpload?: ActivityFileAnswerDto): boolean {
+        const answer = answerBeforeUpload || this.block.answer;
+
+        if (this.isMaxFileSizeExceeded(answer)) {
+            this.result = 'The file size is too big';
+            return false;
+        }
+
+        if (this.isNotAllowedFileType(answer)) {
+            this.result = 'The file mime-type is not allowed';
+            return false;
+        }
+
+        this.result = null;
+        return true;
+    }
+
+    private isMaxFileSizeExceeded(answer: ActivityFileAnswerDto): boolean {
+        return this.block.maxFileSize && answer && answer.fileSize > this.block.maxFileSize;
```

Comment:
#minor answer && answer.fileSize => answer?.fileSize
same for the line 33

Reply:
Ok, fixed at 4e8150675e0c6ee9724cd52ec02d8f73120cc52a